tests/jax/test_distributed_layernorm_mlp.py

```diff
@@ -0,0 +1,266 @@
+                                    FP8Helper.AMAX_HISTORY_LEN), jnp.float32)
+        fp8_metas_scale = jnp.ones((FP8Helper.NUM_META_PER_GEMM * 2, 1), jnp.float32)
+        fp8_metas_scale_inv = jnp.ones((FP8Helper.NUM_META_PER_GEMM * 2, 1), jnp.float32)
+
+        inputs = [x, gamma, k1, k2, b1, b2] = \
+            self.generate_inputs(input_shape, activation_type, use_bias, dtype)
+        inputs = [*inputs, fp8_max, fp8_metas_amax, fp8_metas_scale, fp8_metas_scale_inv]
+        static_inputs = [layernorm_type,
+                         activation_type,
+                         use_bias]
+        value_and_grad_func = jax.value_and_grad(self.layernorm_fp8_mlp_prim_func,
+                                                 argnums=range(len(inputs)))
+
+        # Single GPU
+        single_jitter = jax.jit(value_and_grad_func,
+                                static_argnums=range(len(inputs),
+                                                     len(static_inputs)+len(inputs)))
+        with fp8_autocast(enabled=True):
+            single_fwd, single_grads = single_jitter(*inputs, *static_inputs)
+
+        # Multi GPUs
+        devices = np.asarray(jax.devices()[:device_count]).reshape(*mesh_shape)
+        mesh = Mesh(devices, mesh_axes)
+        with mesh, fp8_autocast(enabled=True, mesh_resource=mesh_resource):
+            k1_sharding = NamedSharding(mesh, PartitionSpec('fsdp', None, 'tp'))
+            k2_sharding = NamedSharding(mesh, PartitionSpec('tp', 'fsdp'))
+            k1_ = jax.device_put(k1, k1_sharding)
+            k2_ = jax.device_put(k2, k2_sharding)
+            if use_bias:
+                b1_sharding = NamedSharding(mesh, PartitionSpec(None, 'tp'))
+                b1_ = jax.device_put(b1, b1_sharding)
+            else:
+                b1_sharding = b1_ = None
+            multi_inputs = [*inputs[:2], k1_, k2_, b1_, *inputs[5:]]
+
+            # Position ref for sharding pspec lists
+            #   x, gamma, k1, k2, b1,
+            #   b2, fp8_max, fp8_metas_amax, fp8_metas_scale, fp8_metas_scale_inv
+            in_shardings = (None, None, k1_sharding, k2_sharding, b1_sharding,
+                            None, None, None, None, None)
+            out_shardings = (None, (None, None, k1_sharding, k2_sharding, b1_sharding,
+                                    None, None, None, None, None))
+
+            multi_jitter = jax.jit(
+                value_and_grad_func,
+                in_shardings=in_shardings,
+                out_shardings=out_shardings,
+                static_argnums=range(len(multi_inputs),
+                                     len(static_inputs)+len(multi_inputs)+1))   # +1 for multi_gpus
+
+            multi_fwd, multi_grads = multi_jitter(*multi_inputs, *static_inputs, True)
+
+        assert_allclose(multi_fwd, single_fwd, dtype=dtype)
+        for i in range(len(inputs)):
+            if multi_grads[i] is not None:
+                assert_allclose(multi_grads[i], single_grads[i], dtype=dtype)
```

Comment:
When wrapping a `assert_xxx` into a for loop, it is important to include an error message. so that we can quick identify which tensors have issue. This helps us quickly identify which tensors might have issues. Could you add an error message to the assert_allclose arguments? For example, something like err_msg=f'{grads[i]} is not close'.